### PR TITLE
Eio port

### PR DESCRIPTION
**Pull request by talex5**
This switches capnp-rpc from Lwt to Eio. One particularly nice side effect of this is that `Service.return_lwt` has gone, as there is no distinction now between concurrent and non-concurrent service methods.

Notes:

- Some of the "unix" functions can be moved into the core library with Eio.
- Mirage support is gone. The main library should now work with both Unix and unikernels.

This commit aims to keep the diff (relatively) small. Several more things are needed after this is merged. In particular:
- Performance improvements (such as sending multiple messages at once).
- Handling leaked refs correctly. We need to schedule a callback on our owning domain's event loop from the finalizer. Eio doesn't currently provide a way to do this.

### CODE REVIEW

**Comment by tmcgilchrist** [comment]
Can we keep the LWT version available for OCaml 4.14? We will need to move services to OCaml 5 and EIO over a longer period of time, both for platform support reasons and just because it will take some time to port things over.

**Comment by talex5** [comment]
> Can we keep the LWT version available for OCaml 4.14?

The existing release will continue to work there. capnp-rpc is pretty stable, so I'm not expecting any major new features that would need back-porting.

**Comment by tmcgilchrist** [comment]
We'll take a release of this when it's ready. [clarke](https://github.com/ocurrent/clarke) and [solver-service](https://github.com/ocurrent/solver-service/pull/71) could use it.

File: unix/network.ml [hunk 1]
```diff
@@ -89,13 +89,19 @@ let connect net ~sw ~secret_key (addr, auth) =
...
     end;
     Tls_wrapper.connect_as_client socket secret_key auth
   | exception ex ->
     error "@[<v2>Network connection for %a failed:@,%a@]" Location.pp addr Fmt.exn ex
 
 let accept_connection ~secret_key flow =
+  Eio_unix.Resource.fd_opt flow
+  |> Option.iter (fun fd ->
+      Eio_unix.Fd.use_exn "TCP_NODELAY" fd @@ fun fd ->
+      Unix.setsockopt fd Unix.TCP_NODELAY true
```
Comment (patricoferris) [inline unix/network.ml:103]: We might need to wrap this to catch

```ocaml
Unix.Unix_error(Unix.EOPNOTSUPP, "setsockopt", "")
```

Stumbled across this new change when using this branch on macOS with a unix domain socket.

**Comment by talex5** [comment]
Everything here got merged with #284 and #287, except for disabling Nagle's algorithm, which is probably no longer necessary. At least, it no longer seems to affect the benchmark results.

**Pull request closed without merging**